services/mlmodel/client.go

```diff
@@ -41,20 +45,141 @@ func NewClientFromConn(
 	return c, nil
 }
 
-func (c *client) Infer(ctx context.Context, input map[string]interface{}) (map[string]interface{}, error) {
+func (c *client) Infer(ctx context.Context, tensors ml.Tensors, input map[string]interface{}) (ml.Tensors, map[string]interface{}, error) {
 	inProto, err := structpb.NewStruct(input)
 	if err != nil {
-		return nil, err
+		return nil, nil, err
+	}
+	tensorProto, err := TensorsToProto(tensors)
+	if err != nil {
+		return nil, nil, err
 	}
 	resp, err := c.client.Infer(ctx, &pb.InferRequest{
-		Name:      c.name,
-		InputData: inProto,
+		Name:         c.name,
+		InputTensors: tensorProto,
+		InputData:    inProto,
```

Comment:
Right now, the C++ SDK will reject an `InferRequest` that has both `InputTensors` and `InputData` set, so I'd suggest not setting `InputData` here.

Reply:
changed

Reply:
in the sense that I check to make sure that only one or the other is nil